tests/ibc-hooks/xcs_cw20_test.go

```diff
@@ -0,0 +1,179 @@
+	chainB := suite.GetChain(ChainB)
+
+	osmosisApp := chainB.GetOsmosisApp()
+	contractKeeper := wasmkeeper.NewDefaultPermissionKeeper(osmosisApp.WasmKeeper)
+
+	if len(memo) == 0 {
+		memo = "\"\""
+	}
+
+	transferMsg := fmt.Sprintf(`{"channel": "%s", "remote_address": "%s", "memo": %s}`, path.EndpointB.ChannelID, receiver.String(), memo)
+	transferMsgBase64 := base64.StdEncoding.EncodeToString([]byte(transferMsg))
+
+	ctx := chainB.GetContext()
+	_, err := contractKeeper.Execute(
+		ctx,
+		cw20Addr,
+		chainB.SenderAccount.GetAddress(),
+		[]byte(fmt.Sprintf(`{"send": {"contract": "%s", "amount": "%s", "msg": "%s"}}`, cw20ics20Addr.String(), amount, transferMsgBase64)),
+		nil,
+	)
+	suite.Require().NoError(err)
+
+	// Move forward one block
+	chainB.NextBlock()
+	chainB.Coordinator.IncrementTime()
+
+	// Update both clients
+	err = path.EndpointB.UpdateClient()
+	suite.Require().NoError(err)
+	err = path.EndpointA.UpdateClient()
+	suite.Require().NoError(err)
+
+	events := ctx.EventManager().Events()
+	packet, err := ibctesting.ParsePacketFromEvents(events)
+	suite.Require().NoError(err)
+	result, ack := suite.RelayPacket(packet, CW20toA)
+	suite.Require().Contains(string(ack), "result")
+	return result, ack
+}
+
+func (suite *HooksTestSuite) setupCW20PoolAndRoutes(chain *osmosisibctesting.TestChain, swaprouterAddr sdk.AccAddress, cw20IbcDenom string, amount sdk.Int) {
+	osmosisAppA := chain.GetOsmosisApp()
+	poolId := suite.CreateIBCPoolOnChain(ChainA, cw20IbcDenom, "stake", amount)
+
+	// create a swap route for that token / poolId in both directions
+	msg := fmt.Sprintf(`{"set_route":{"input_denom":"%s","output_denom":"%s","pool_route":[{"pool_id":"%v","token_out_denom":"%s"}]}}`,
+		cw20IbcDenom, "stake", poolId, "stake")
+	contractKeeper := wasmkeeper.NewDefaultPermissionKeeper(osmosisAppA.WasmKeeper)
+	_, err := contractKeeper.Execute(chain.GetContext(), swaprouterAddr, chain.SenderAccount.GetAddress(), []byte(msg), sdk.NewCoins())
+	suite.Require().NoError(err)
+	msg = fmt.Sprintf(`{"set_route":{"input_denom":"%s","output_denom":"%s","pool_route":[{"pool_id":"%v","token_out_denom":"%s"}]}}`,
+		"stake", cw20IbcDenom, poolId, cw20IbcDenom)
+	_, err = contractKeeper.Execute(chain.GetContext(), swaprouterAddr, chain.SenderAccount.GetAddress(), []byte(msg), sdk.NewCoins())
+	suite.Require().NoError(err)
+}
+
+func (suite *HooksTestSuite) getCW20Balance(chain *osmosisibctesting.TestChain, cw20Addr, addr sdk.AccAddress) sdk.Int {
+	queryMsg := fmt.Sprintf(`{"balance":{"address":"%s"}}`, addr)
+	res := chain.QueryContractJson(&suite.Suite, cw20Addr, []byte(queryMsg))
+	x, ok := sdk.NewIntFromString(res.Get("balance").String())
+	if !ok {
+		panic("could not parse balance")
+	}
+	return x
+}
+
+// Test that the cw20-ics20 contract can be instantiated and used to send tokens to chainA
+func (suite *HooksTestSuite) TestCW20ICS20() {
+	suite.Require().Equal("stake", sdk.DefaultBondDenom) // hardcodiing "stake" for simplicity
```

Comment:
Not sure I understand, wouldn't it just be better to utilize sdk.DefaultBondDenom instead of explicitly using stake?

Reply:
you're right. There were other places where "stake" was being used and I was too lazy to change it. Fixed now